Sources/PackageModel/ResolvedToolset.swift

```diff
@@ -0,0 +1,482 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift open source project
+//
+// Copyright (c) 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Basics
+
+import class TSCBasic.Process
+
+#if os(Windows)
+private let hostExecutableSuffix = ".exe"
+#else
+private let hostExecutableSuffix = ""
+#endif
+
+public struct ResolvedToolset {
+    typealias SwiftCompilers = (compile: AbsolutePath, manifest: AbsolutePath)
+
+    public struct ToolProperties {
+        public let path: AbsolutePath
+        public let extraCLIOptions: [String]
+    }
+
+    public let effectiveTriple: Triple
+
+    public let cCompiler: ToolProperties
+    public let cxxCompiler: ToolProperties
+    public let debugger: ToolProperties
+    public let librarian: ToolProperties
+    public let swiftCompiler: ToolProperties
+    // TODO: Swift compiler for manifests?
+    public let swiftPluginServer: ToolProperties?
+    public let testRunner: ToolProperties?
+    // Note: XCBuild is intentionally not part of the resolved toolset
+
+    public var swiftInterpreter: ToolProperties {
+        return ToolProperties(
+            path: self.swiftCompiler.path.parentDirectory.appending("swift" + hostExecutableSuffix),
+            extraCLIOptions: self.swiftCompiler.extraCLIOptions)
+    }
+
+    public enum SearchStrategy {
+        case `default`
+        case custom(searchPaths: [AbsolutePath], useXcrun: Bool = true)
+    }
+
+    /// Search paths from the PATH environment variable.
+    let envSearchPaths: [AbsolutePath]
+
+    /// Only use search paths, do not fall back to `xcrun`.
+    let useXcrun: Bool
+
+    init(
+        sdkRootPath: AbsolutePath?,
+        toolset: Toolset,
+        triple: Triple?, // FIXME: This sucks because there's still an opportunity to misuse it and compute the triple multiple times.
```

Comment:
ditto for being explicit about host/target triple in names

Reply:
see above